src/node_mime.cc

```diff
@@ -0,0 +1,370 @@
+#include <string>
+#include "env.h"
+#include "node.h"
+#include "node_mime.h"
+#include "node_internals.h"
+#include "util-inl.h"
+
+namespace node {
+
+namespace mime {
+
+using v8::Array;
+using v8::Context;
+using v8::Function;
+using v8::FunctionCallbackInfo;
+using v8::FunctionTemplate;
+using v8::HandleScope;
+using v8::Isolate;
+using v8::Local;
+using v8::Object;
+using v8::String;
+using v8::Value;
+
+
+// utilities for
+namespace {
+
+bool is_ascii_upper_alpha(char c) {
+  return c >= '\x41' && c <= '\x5A';
+}
+
+bool is_ascii_lower_alpha(char c) {
+  return c >= '\x61' && c <= '\x7A';
+}
+
+bool is_ascii_digit(char c) {
+  return c >= '\x30' && c <= '\x39';
+}
+
+bool is_ascii_alpha(char c) {
+  return is_ascii_upper_alpha(c) || is_ascii_lower_alpha(c);
+}
+
+bool is_ascii_alphanumeric(char c) {
+  return is_ascii_digit(c) || is_ascii_alpha(c);
+}
+
+bool is_http_token(char c) {
+  return c == '\x21' ||
+      c == '\x23' ||
+      c == '\x24' ||
+      c == '\x25' ||
+      c == '\x26' ||
+      c == '\x27' ||
+      c == '\x2A' ||
+      c == '\x2B' ||
+      c == '\x2D' ||
+      c == '\x2E' ||
+      c == '\x5E' ||
+      c == '\x5F' ||
+      c == '\x60' ||
+      c == '\x7C' ||
+      c == '\x7E' ||
+      is_ascii_alphanumeric(c);
+}
+
+bool is_ascii_whitespace(char c) {
+  return c == '\x09' ||
+      c == '\x0A' ||
+      c == '\x0C' ||
+      c == '\x0D' ||
+      c == '\x20';
+}
+
+bool is_http_quoted_string_token(char c) {
+  return c == '\x09' ||
+      (c >= '\x20' && c <= '\x7E') ||
+      (c >= '\x80' && c <= '\xFF');
```

Comment:
For legibility reasons, I'd replace the hexadecimal escapes with their printable counterparts where possible.

Reply:
fixed